test/distributed/test_c10d_nccl.py

```diff
@@ -438,6 +438,35 @@ def test_restart_pg(self):
         with self.assertRaises(ValueError):
             dist.all_reduce(t1)
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_cuda_event_cache_mthd_race(self):
+        # This unit test is to test the case when the collective is launched in
+        # a side thread and the thread dies before the cache has been fully recycled.
+        # More details can be found in this issue: https://github.com/pytorch/pytorch/issues/143470.
+        import threading
+
+        # initiate collectives here
+        def init_collective_task(t):
+            dist.all_reduce(t)
+            dist.all_reduce(t)
+            dist.all_reduce(t)
+
+        os.environ["TORCH_NCCL_CUDA_EVENT_CACHE"] = "1"
```

Comment:
nit: caution when setting env in unit tests -- it can carry over to other unit tests if the process is not restarted -- like we did for `test_c10d_ops_nccl.py` to reuse PGs. `test_c10d_nccl` is fine, but can be refactored in future.

Reply:
I can add a ENV reset, lol.

Reply:
Hmm since we launch it anyway so I think this should be fine now. And we want to remove the ENV maybe before next PT release.